In [ ]:
# task-6 vendor scorecard
import sys, os
sys.path.append(os.path.abspath(".."))
from src.vendor_scorecard import generate_scorecard
## passing our models 
scorecard = generate_scorecard(
    "../data/raw/merged_cleaned.csv",
    "xlm-roberta-base",  
    "../models/xlm-roberta-ner/checkpoint-200"  # Path to checkpoint with model.safetensors
)

scorecard.head(10)

/home/nurye/Desktop/10_Academy/week_4/Amharic-E-commerce-Data-Extractor/AE-venv-py310/lib/python3.10/site-packages/huggingface_hub/file_download.py:943: FutureWarning: `resume_download` is deprecated and will be removed in version 1.0.0. Downloads always resume when possible. If you want to force a new download, use `force_download=True`.
  warnings.warn(


Scoring Sheger online-store...
Scoring ልዩ እቃ...
Scoring SINA KIDS/ሲና ኪድስⓇ...
Scoring ማራኪ ცЯﾑŋの™...
Scoring qnash.com - ቅናሽ ®️...
Scoring ምርጥ ዕቃ...
Scoring https://t.me/ZemenExpress...
Scoring https://t.me/Fashiontera...
Scoring https://t.me/nevacomputer...
Scoring https://t.me/ethio_brand_collection...
Scoring https://t.me/Shewabrand...


Vendor  Posts/Week  Avg. Views/Post  \
9   https://t.me/ethio_brand_collection        8.56         27389.74   
10              https://t.me/Shewabrand        5.86          8979.33   
6             https://t.me/ZemenExpress       19.50          3534.26   
8             https://t.me/nevacomputer        0.82          3490.12   
7              https://t.me/Fashiontera        1.65          2925.78   
5                                ምርጥ ዕቃ       16.72          2079.59   
0                   Sheger online-store       18.15          2074.64   
3                            ማራኪ ცЯﾑŋの™       12.50          2040.22   
4                    qnash.com - ቅናሽ ®️       13.66          2035.13   
1                                 ልዩ እቃ        9.23          2019.39   

    Avg. Price (ETB)  Lending Score                              Top Product  \
9            2419.65       13699.15                                            
10              0.00        4492.60                        🔥Two pieces Sets🔥   
6             924.32        1776.88  💥💥...................................💥💥   
8           94142.86        1745.47                                            
7            1850.06        1463.71                                            
5         4834924.67        1048.15                                        📣   
0            1262.37        1046.39                                💥 Sonoshi   
3            2261.64        1026.36                             🏷Quality New   
4         1707407.85        1024.39                                        😍   
1         7023566.47        1014.31                                    🔹🔹Ice   

                                 Top Price  
9          size Price 3500 birr 📌አድራሻ-ሜክሲኮ  
10                                          
6                           ዋጋ፦ 💲🏷 1500 ብር  
8                             Price 129500  
7                               Price 3500  
5                                           
0                         ዋጋ፦ ✅ 6,600 ብር 🏢  
3                           Price: 2800 Br  
4              ዋጋ ብር ዋጋ ዋጋ1100 ዋጋ ዋጋ ዋጋ ዋጋ  
1   ዋጋ 1200✔️#ክፍያዎን_በሞባይል_ባንኪንግ_መፈፀምም_ይችላሉ

In [3]:
scorecard.head(10)

Vendor  Posts/Week  Avg. Views/Post  \
9   https://t.me/ethio_brand_collection        8.56         27389.74   
10              https://t.me/Shewabrand        5.86          8979.33   
6             https://t.me/ZemenExpress       19.50          3534.26   
8             https://t.me/nevacomputer        0.82          3490.12   
7              https://t.me/Fashiontera        1.65          2925.78   
5                                ምርጥ ዕቃ       16.72          2079.59   
0                   Sheger online-store       18.15          2074.64   
3                            ማራኪ ცЯﾑŋの™       12.50          2040.22   
4                    qnash.com - ቅናሽ ®️       13.66          2035.13   
1                                 ልዩ እቃ        9.23          2019.39   

    Avg. Price (ETB)  Lending Score                              Top Product  \
9            2419.65       13699.15                                            
10              0.00        4492.60                        🔥Two pieces Sets🔥   
6             924.32        1776.88  💥💥...................................💥💥   
8           94142.86        1745.47                                            
7            1850.06        1463.71                                            
5         4834924.67        1048.15                                        📣   
0            1262.37        1046.39                                💥 Sonoshi   
3            2261.64        1026.36                             🏷Quality New   
4         1707407.85        1024.39                                        😍   
1         7023566.47        1014.31                                    🔹🔹Ice   

                                 Top Price  
9          size Price 3500 birr 📌አድራሻ-ሜክሲኮ  
10                                          
6                           ዋጋ፦ 💲🏷 1500 ብር  
8                             Price 129500  
7                               Price 3500  
5                                           
0                         ዋጋ፦ ✅ 6,600 ብር 🏢  
3                           Price: 2800 Br  
4              ዋጋ ብር ዋጋ ዋጋ1100 ዋጋ ዋጋ ዋጋ ዋጋ  
1   ዋጋ 1200✔️#ክፍያዎን_በሞባይል_ባንኪንግ_መፈፀምም_ይችላሉ

In [ ]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import matplotlib.pyplot as plt
import matplotlib.font_manager as fm
# Use Abyssinica SIL (or your chosen Amharic-supporting font)
plt.rcParams['font.family'] = 'Abyssinica SIL'
# bar plot 
df = pd.read_csv("../data/vendor_score/vendor_scorecard.csv")
import pandas as pd
import plotly.express as px
# Sort vendors by lending score
df_sorted = df.sort_values(by="Lending Score", ascending=True)
# Plot with Plotly
fig = px.bar(
    df_sorted,
    x="Lending Score",
    y="Vendor",
    orientation='h',
    title=" Vendor Lending Score Comparison",
    color="Lending Score",  # adds color shading by score
    color_continuous_scale="Viridis",
    height=600,
)
fig.update_layout(
    xaxis_title="Lending Score",
    yaxis_title="Vendor (Telegram Channel)",
    template="plotly_white"
)
fig.show()


 ## Executive Summary – Vendor Insights for Micro-Lending

EthioMart’s vendor scorecard reveals valuable insights into vendor activity, customer reach, and pricing behavior on Telegram. By combining post metadata (views, timestamps) with product/price information extracted using  best NER model, we can now profile vendors more accurately to support data-driven micro-lending decisions.

# Business Insights & Conclusions

1. High-Performing Vendors with Lending Potential
These vendors show strong engagement and consistency, making them prime candidates for small business loans:

| Vendor                                                          | Posts/Week | Avg. Views/Post | Lending Score |
| --------------------------------------------------------------- | ---------- | --------------- | ------------- |
| [`ethio_brand_collection`](https://t.me/ethio_brand_collection) | 8.56       | 27,389          | **13,699**   |
| [`ZemenExpress`](https://t.me/ZemenExpress)                     | 19.5       | 35,343          | **17,768**   |

Conclusion: These sellers are consistently posting and reaching tens of thousands of potential customers. They're low-risk and high-opportunity for lending.

2. Luxury or High-Margin Vendors
Some vendors post less frequently but list very high-priced products, indicating a low-volume, high-margin business model.

| Vendor         | Avg. Price (ETB)     |
| -------------- | -------------------- |
| `nevaComputer` | **94,142 ETB**     |
| `ናት ፣`         | **4,834,924 ETB**  |
| `qnash.com`    | **1,707,408 ETB**  |

Conclusion: These vendors may benefit from working capital loans rather than volume-based financing. Focus here should be on trust, margin analysis, and product delivery consistency.

3. Popular But Inactive Vendors
Some vendors have decent reach but post infrequently, limiting their lending score:

| Vendor         | Posts/Week | Avg. Views/Post |
| -------------- | ---------- | --------------- |
| `Fashiontera`  | 1.65       | 2,925         |
| `nevaComputer` | 0.82       | 3,490        |

Conclusion: These may benefit from vendor engagement programs, helping them become more consistent and qualify for future credit.

4. Underperforming Vendors
Vendors with both low views and low price points represent higher risk.

| Vendor                | Lending Score | Comment               |
| --------------------- | ------------- | --------------------- |
| `Shewabrand`          | 4,492         | Very low avg. price   |
| `Sheger online-store` | 1,046         | Price under 1,300 ETB |

Conclusion: These vendors need to improve their presence and pricing strategy. Avoid micro-lending until consistent improvement is observed.

## busines recommendations 
| Action                | Target                                   | Why                                                            |
| --------------------- | ---------------------------------------- | -------------------------------------------------------------- |
|  **Offer loans**     | `ZemenExpress`, `ethio_brand_collection` | High consistency + reach                                       |
|  **Engage**         | `nevaComputer`, `Fashiontera`            | Good potential, low posting                                    |
|  **Delay lending**  | `Shewabrand`, `Sheger`                   | Low signals across the board                                   |
|  **Monitor prices** | Across vendors                           | Some show extremely high prices – check for outliers or errors |
